config/config.go

```diff
@@ -104,6 +104,11 @@ func (a *Auth) Obfuscate() {
 	a.CAFile = "xxx"
 }
 
+type AddonComponentStatus struct {
```

Comment:
Do you think is necessary to add a nest struct at this level ?
I mean, why not to keep the IsCore as the current place and add the HealthCheckUrl at that level ?
If present is used, if not the behaviour is like the previous approach.

I comment this, because, config is growing and growing and not sure if adding more nest elements add a little bit of complexity to read it.